realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -26,11 +26,15 @@
 #include "jni_util/java_global_weak_ref.hpp"
 #include "jni_util/java_method.hpp"
 #include "jni_util/java_class.hpp"
+#include "jni_util/java_exception_thrower.hpp"
 
 using namespace realm;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
+static const char* PK_CONSTRAINT_EXCEPTION_CLASS = "io/realm/exceptions/RealmPrimaryKeyConstraintException";
```

Comment:
Would it make sense to move all Java exception types into the java_exception_thrower class instead of having them scattered around?